cmd/shipperctl/configurator/cluster.go

```diff
@@ -280,3 +294,175 @@ func loadKubeConfig(context, kubeConfigFile string) (*rest.Config, error) {
 
 	return clientConfig.ClientConfig()
 }
+
+func (c *Cluster) CreateCertificateSigningRequest(csr []byte) error {
+	certificateSigningRequest := &certificatesv1beta1.CertificateSigningRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: shipperCSRName,
+		},
+		Spec: certificatesv1beta1.CertificateSigningRequestSpec{
+			Request: csr,
+			Groups:  []string{"system:authenticated"},
+			Usages: []certificatesv1beta1.KeyUsage{
+				certificatesv1beta1.UsageServerAuth,
+				certificatesv1beta1.UsageDigitalSignature,
+				certificatesv1beta1.UsageKeyEncipherment,
+			},
+		},
+	}
+
+	_, err := c.KubeClient.CertificatesV1beta1().CertificateSigningRequests().Create(certificateSigningRequest)
+	return err
+}
+
+func (c *Cluster) ApproveShipperCSR() error {
+	csr, err := c.KubeClient.CertificatesV1beta1().CertificateSigningRequests().Get(shipperCSRName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	approvedCondition := certificatesv1beta1.CertificateSigningRequestCondition{
+		Type:    certificatesv1beta1.CertificateApproved,
+		Reason:  "ShipperctlApprove",
+		Message: "Automatically approved by shipperctl",
```

Comment:
Not sure if this makes sense, but would it be helpful if we include the timestamp when it was approved?

Reply:
That one is automatically added by Kubernetes. I ran `shipperctl` locally, and here is what I see in the `conditions` of the CertificateSigningRequest object:

```
  conditions:
  - lastUpdateTime: "2019-04-05T13:35:25Z"
    message: Automatically approved by shipperctl
    reason: ShipperctlApprove
    type: Approved
```

Reply:
Good point, makes sense.